packages/main/cypress/specs/MultiInput.cy.tsx

```diff
@@ -69,4 +88,1235 @@ describe("MultiInput Web Component", () => {
+			.as("respPopover");
+
+		cy.get("ui5-multi-input")
+			.shadow()
+			.find("input")
+			.as("input");
+
+		cy.get("@respPopover")
+			.should("have.attr", "open");
+
+		cy.get("@tokenizer")
+			.shadow()
+			.find("ui5-li")
+			.as("tokenLi");
+
+		cy.get("@tokenLi")
+			.should("have.focus");
+
+		cy.get("ui5-token")
+			.should("have.attr", "selected");
+	})
+	it("should open responsive popover on click on single truncated token and close and deselect the token on clicking it again", () => {
+
+		cy.get("ui5-token")
+			.click();
+
+		cy.get("ui5-token")
+			.should("have.focus");
+
+		cy.get("ui5-token")
+			.should("not.have.attr", "selected");
+
+		cy.get("@respPopover")
+			.should("not.have.property", "open");
+	});
+	it("should close truncation popover and deselect selected tokens when clicked outside the component", () => {
+		cy.get("#dummyBtn")
+			.click();
+
+		cy.get("ui5-token")
+			.should("not.have.focus");
+
+		cy.get("ui5-token")
+			.should("not.have.attr", "selected");
+
+		cy.get("@respPopover")
+			.should("not.have.property", "open");
+	})
+	it("should close truncation popover and deselect selected tokens when clicked in input field", () => {
+		cy.get("@input")
+			.click();
+
+		cy.get("ui5-token")
+			.should("not.have.focus");
+
+		cy.get("ui5-token")
+			.should("not.have.attr", "selected");
+
+		cy.get("@respPopover")
+			.should("not.have.property", "open");
+	});
+	it("should truncate token when a long token is added", () => {
+		cy.get("ui5-multi-input")
+			.then(multiInput => {
+				multiInput[0].addEventListener("ui5-token-delete", handleTokenDelete);
+
+				multiInput[0].addEventListener("keydown", (event: KeyboardEvent) => {
+					const inputElement = multiInput[0].shadowRoot.querySelector("input") as HTMLInputElement;
+					if (event.key === "Enter" && inputElement && inputElement.value) {
+						const token = createTokenFromText(inputElement.value);
+						multiInput[0].appendChild(token);
+						inputElement.value = "";
+					}
+				});
+			});
+
+		cy.get("@tokenLi")
+			.shadow()
+			.find("ui5-button")
+			.as("deleteIcon");
+
+		cy.get("@deleteIcon")
+			.click();
+
+		cy.get("@input")
+			.should("be.focused");
+
+		cy.get("@input")
+			.click();
+
+		cy.get("@input")
+			.type("Officia enim ullamco sunt sunt nisi ullamco cillum velit ullamco cillum velit ullamco cillum");
+
+		cy.get("@input")
+			.realPress("Enter");
+
+		cy.get("ui5-token")
+			.click();
+
+		cy.get("@respPopover")
+			.should("have.attr", "open");
+
+		cy.get("ui5-token")
+			.should("have.attr", "single-token");
+
+		cy.get("ui5-token")
+			.should("have.attr", "selected");
+	});
+	it("should not throw exception when MI with 1 token is added to the page", () => {
+		cy.mount(
+			<>
+			<Button id="add-single-token">Add MultiInput with single token</Button>
+			<div id="container"></div>
+			</>
+		);
+
+		cy.get("ui5-button")
+			.then(button => {
+				button[0].addEventListener("click", () => {
```

Comment:
this could be reworked by querying the container and in then to execute the callback